src/Bicep.Core/Emit/EmitLimitationCalculator.cs

```diff
@@ -93,5 +96,153 @@ public static EmitLimitationInfo Calculate(SemanticModel semanticModel)
 
             return moduleScopeData.ToImmutableDictionary();
         }
+
+        private static IEnumerable<Diagnostic> DetectDuplicateNames(SemanticModel semanticModel, ImmutableDictionary<ResourceSymbol, ResourceSymbol?> resourceScopeData, ImmutableDictionary<ModuleSymbol, ScopeHelper.ScopeData> moduleScopeData)
+        {
+            // This method only checks, if in one deployment we do not have 2 or more resources with this same name in one deployment to avoid template validation error
+            // This will not check resource constraints such as necessity of having unique virtual network names within resource group
+
+            var duplicateResources = semanticModel.Root.AllDeclarations
+                .OfType<ResourceSymbol>()
```

Comment:
Use `semanticModel.Root.ResourceDeclarations` to access resource symbols

Reply:
done.